src/CTA.WebForms2Blazor/Services/LifecycleManagerService.cs

```diff
@@ -0,0 +1,228 @@
+        }
+
+        public static bool ContentIsPreHandle(WebFormsAppLifecycleEvent lcEvent)
+        {
+            return ((int)lcEvent) < (int)Constants.FirstPostHandleEvent;
+        }
+
+        public static WebFormsAppLifecycleEvent? CheckMethodApplicationLifecycleHook(MethodDeclarationSyntax methodDeclaration)
+        {
+            if (methodDeclaration.HasEventHandlerParameters())
+            {
+                var methodName = methodDeclaration.Identifier.ToString();
+
+                foreach (int value in Enum.GetValues(typeof(WebFormsAppLifecycleEvent)))
+                {
+                    var currentEvent = (WebFormsAppLifecycleEvent)value;
+
+                    if (methodName.Equals(ApplicationLifecycleHookMethodPrefix + currentEvent.ToString()))
+                    {
+                        return currentEvent;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        public static WebFormsPageLifecycleEvent? CheckMethodPageLifecycleHook(MethodDeclarationSyntax methodDeclaration)
+        {
+            if (methodDeclaration.HasEventHandlerParameters())
+            {
+                var methodName = methodDeclaration.Identifier.ToString();
+
+                foreach (int value in Enum.GetValues(typeof(WebFormsPageLifecycleEvent)))
+                {
+                    var currentEvent = (WebFormsPageLifecycleEvent)value;
+
+                    if (methodName.Equals(PageLifecycleHookMethodPrefix + currentEvent.ToString()))
+                    {
+                        return currentEvent;
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        public static BlazorComponentLifecycleEvent GetEquivalentComponentLifecycleEvent(WebFormsPageLifecycleEvent lcEvent)
+        {
+            var lcEventIndex = (int)lcEvent;
+
+            if (lcEventIndex < (int)Constants.FirstOnInitializedEvent)
+            {
+                return BlazorComponentLifecycleEvent.SetParametersAsync;
+            }
+            else if (lcEventIndex < (int)Constants.FirstOnParametersSetEvent)
+            {
+                return BlazorComponentLifecycleEvent.OnInitialized;
+            }
+            else if (lcEventIndex < (int)Constants.FirstOnAfterRenderEvent)
+            {
+                return BlazorComponentLifecycleEvent.OnParametersSet;
+            }
+            else if (lcEventIndex < (int)Constants.FirstDisposeEvent)
+            {
+                return BlazorComponentLifecycleEvent.OnAfterRender;
+            }
+            else
+            {
+                return BlazorComponentLifecycleEvent.Dispose;
+            }
+        }
+
+        private Task<bool> WaitForAllMiddlewareRegistered(CancellationToken token)
+        {
+            var source = new TaskCompletionSource<bool>();
+
+            if (_numMiddlewareSourcesProcessed >= _expectedMiddlewareSources)
```

Comment:
Is there a reason why this one doesn't mind if there are more processed middleware than line 54 where we specifically throw an error for processing more than the expected number?

Reply:
Originally, I was throwing an exception on line 54 because I wanted to error out converters that aren't expected for processing. If we threw an error for the invalid state it would cause any converter that uses this service to error out as well. Technically the state is still valid though, only it won't have the data from the unexpected converters

Reply:
Nope, that's fine. I think the current coverage is good. Thanks.